lib/aws/rails/middleware/ebs_sqs_active_job_middleware.rb

```diff
@@ -85,7 +85,15 @@ def sent_from_docker_host?(request)
       end
 
       def app_runs_in_docker_container?
-        @app_runs_in_docker_container ||= `[ -f /proc/1/cgroup ] && cat /proc/1/cgroup` =~ /docker/
+        @app_runs_in_docker_container ||= in_docker_containr_with_cgroup1? || in_docker_containr_with_cgroup2?
```

Comment:
Is there a better way other than cgroups (i.e. `dockerenv` file)? Based on my searching, it seems to be the best, but all options don't seem great. 

I'm open to re-evaluating this.

Reply:
I have looked for a way to detect if a process is running inside a docker container, 
but there doesn't seem to be an official and portable way to detect it.

Maybe reading /proc/1/cgrops and /proc/self/mountinfo is the best workaround now.

References:
* https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/6694
* https://github.com/containerd/containerd/issues/8185
* https://github.com/opencontainers/runtime-spec/issues/1105
* https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/CgroupV2ContainerIdExtractor.java
* https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/instrumentation/resources/library/src/main/java/io/opentelemetry/instrumentation/resources/CgroupV1ContainerIdExtractor.java